### PR TITLE
Rewrite sidebar

### PR DESCRIPTION
**Pull request by ajronn**
Change layout
Add open/close modal logic

### CODE REVIEW

File: src/modules/main/Main.tsx [hunk 1]
```diff
@@ -21,11 +21,15 @@ const TemplateDocumentation = withLazy(() => import('src/modules/template-docume
 const TemplateManagement = withLazy(() => import('src/modules/template-management'));
 
 const Main = ({ match }: RouteChildrenProps) => {
+
```
Comment (polubis) [inline src/modules/main/Main.tsx:24]: This logic should be handled inside sidebar component

File: src/modules/main/Main.tsx [hunk 2]
```diff
@@ -21,11 +21,15 @@ const TemplateDocumentation = withLazy(() => import('src/modules/template-docume
...
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+  const changeSidebarState = () => setIsSidebarOpen(!isSidebarOpen);
+
   return (
-    <div className={csx.main}>
+    <div className={csx.main} style={{gridTemplateColumns: isSidebarOpen ? "150px 1fr" : "50px 1fr"}}>
```
Comment (polubis) [inline src/modules/main/Main.tsx:29]: Remove this style change - sidebar menu should be fixed / static based on current state.

File: src/modules/main/sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -32,16 +34,28 @@ const getActiveLinkIdx = (pathname: string, links: SidebarLink[]) => {
   });
 };
 
-export const Sidebar = ({ basePath }: SidebarProps) => {
+export const Sidebar = ({ basePath, isSidebarOpen, changeSidebarState }: SidebarProps) => {
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:37]: Remove this props - this should be sidebar inner logic

File: src/modules/main/sidebar/Sidebar.tsx [hunk 4]
```diff
@@ -32,16 +34,28 @@ const getActiveLinkIdx = (pathname: string, links: SidebarLink[]) => {
...
 
   return (
     <aside className={csx.sidebar}>
       <div className={csx.sidebarContent}>
-        <figure className={csx.logo}>
-          <Logo />
+        <figure className={csx.logo} >
+          {isSidebarOpen 
+            ? 
+              <div style={{
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:48]: Avoid inline styles should be replaced with @include row in scsss

File: src/modules/main/sidebar/Sidebar.tsx [hunk 5]
```diff
@@ -32,16 +34,28 @@ const getActiveLinkIdx = (pathname: string, links: SidebarLink[]) => {
...
+                display: 'flex',
+                alignItems: 'center'
+              }}>
+                <img height="30px" src={IMGS_PATH + '/Logo.png'} />
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:52]: Avoid inline css

File: src/modules/main/sidebar/Sidebar.tsx [hunk 6]
```diff
@@ -32,16 +34,28 @@ const getActiveLinkIdx = (pathname: string, links: SidebarLink[]) => {
...
+                <p style={{marginLeft:"10px"}}>Jupi.io</p>
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:53]: Avoid inline css

File: src/modules/main/sidebar/Sidebar.tsx [hunk 7]
```diff
@@ -32,16 +34,28 @@ const getActiveLinkIdx = (pathname: string, links: SidebarLink[]) => {
...
+                <p style={{marginLeft:"10px"}} onClick={()=>changeSidebarState()}><Close /></p>
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:54]: Avoid inline css

File: src/modules/main/sidebar/Sidebar.tsx [hunk 8]
```diff
@@ -55,9 +69,17 @@ export const Sidebar = ({ basePath }: SidebarProps) => {
               to={`${basePath}${path}`}
             >
               <Button>
-                {icon}
-                <span>{label}</span>
+
+                <div style={{
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:73]: Avoid inline css

File: src/modules/main/sidebar/Sidebar.tsx [hunk 9]
```diff
@@ -32,16 +34,28 @@ const getActiveLinkIdx = (pathname: string, links: SidebarLink[]) => {
...
+                display: 'flex',
+                alignItems: 'center'
+              }}>
+                <img height="30px" src={IMGS_PATH + '/Logo.png'} />
+                <p style={{marginLeft:"10px"}}>Jupi.io</p>
+                <p style={{marginLeft:"10px"}} onClick={()=>changeSidebarState()}><Close /></p>
+              </div>
+
+
+            : <Menu onClick={()=>changeSidebarState()} />}
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:58]: change function name to toggleOpen - and add this only inside component. Also add useState variable here and rename to just open, setOpen useState

Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:58]: Replace () => changeSidebarState() - to toggleOpen - in every place without callback

**Review by polubis — CHANGES_REQUESTED** [review]

**Review by polubis — CHANGES_REQUESTED** [review]
This should work like yt sidebar on mobile/tablets but without black backdrop on every devices

File: src/modules/main/sidebar/Sidebar.scss [hunk 10]
```diff
@@ -26,14 +26,7 @@
           border-radius: 0;
           color: $primary;
 
-          & > span {
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.scss:29]: Undo these changes.
Additionally, we also need the labels you removed

**Review by polubis — CHANGES_REQUESTED** [review]
This should work like yt sidebar for mobile/tablet. We dont need backdrop - but menu which will be displayed - should be independent component with fixed pos - rendered by usePortal hook

File: src/modules/main/sidebar/Sidebar.tsx [hunk 11]
```diff
@@ -37,13 +40,20 @@ export const Sidebar = ({ basePath }: SidebarProps) => {
 
   const activeLinkIdx = getActiveLinkIdx(pathname.replace(basePath, ''), sidebarLinks);
 
+  const [open, setOpen] = useState(false);
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:43]: open, setOpen use state should be at the beginning of the component

File: src/modules/main/sidebar/Sidebar.tsx [hunk 12]
```diff
@@ -1,15 +1,18 @@
...
 import DashboardIcon from '@material-ui/icons/Dashboard';
 import ProjectsIcon from '@material-ui/icons/Work';
 import TemplatesIcon from '@material-ui/icons/LibraryBooks';
+import Cancel from '@material-ui/icons/Cancel';
 
 import { Logo } from 'ui';
 
 import { SidebarProps, SidebarLink } from '.';
 
+import {Portal} from './portal';
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:14]: This should be renamed to SidebarPanel

File: src/modules/main/sidebar/portal/Portal.scss [hunk 13]
```diff
@@ -0,0 +1,22 @@
+@import 'styles/mixins';
+
+.portal {
```
Comment (polubis) [inline src/modules/main/sidebar/portal/Portal.scss:3]: .sidebarPanel

File: src/modules/main/sidebar/portal/Portal.scss [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+    border-right: 1px solid rgba($primary, 0.43);
+    background: $surface;
+    position: absolute;
+    z-index: 6;
+    height: 100vh;
+    width: 200px;
+}
+
+.tile {
+    display: flex;
```
Comment (polubis) [inline src/modules/main/sidebar/portal/Portal.scss:14]: @include row instead of flex

File: src/modules/main/sidebar/portal/Portal.tsx [hunk 15]
```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+
+import { NavLink } from 'react-router-dom';
+
+import { Button } from '@material-ui/core';
+import Cancel from '@material-ui/icons/Cancel';
+
+import csx from './Portal.scss'
+import csx2 from '../Sidebar.scss'
```
Comment (polubis) [inline src/modules/main/sidebar/portal/Portal.tsx:9]: Components can only use one style file. If you are using 2 - you have badly planned components

File: src/modules/main/sidebar/portal/Portal.tsx [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+
+const LINK_HEIGHT = 80;
+
+const MARKER_HEIGHT = 30;
+
+export const Portal = ({sidebarLinks, basePath, close}) => {
```
Comment (polubis) [inline src/modules/main/sidebar/portal/Portal.tsx:15]: SidebarPanel same for files

Comment (polubis) [reply src/modules/main/sidebar/portal/Portal.tsx:15]: rename sidebarLinks - to links

File: src/modules/main/sidebar/portal/Portal.tsx [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+                <Cancel fontSize='large' onClick={close}/>
+            </figure>
+
+            <div className={csx2.links}>
+
+            {sidebarLinks.map(({ path, label, icon, exact }) => (
+                <NavLink
+                key={label}
+                exact={exact}
+                activeClassName={csx2.active}
```
Comment (polubis) [inline src/modules/main/sidebar/portal/Portal.tsx:29]: You should create a component - SiedbarLInk which contains repeated parts - f.e styles active, link and this component allows to put children dynamically - to render content inside

File: src/modules/main/sidebar/portal/index.ts [hunk 18]
```diff
@@ -0,0 +1 @@
+export * from './Portal';
```
Comment (polubis) [inline src/modules/main/sidebar/portal/index.ts:1]: rename to SidebarPanel

File: src/modules/main/sidebar/Sidebar.tsx [hunk 19]
```diff
@@ -37,13 +40,20 @@ export const Sidebar = ({ basePath }: SidebarProps) => {
 
   const activeLinkIdx = getActiveLinkIdx(pathname.replace(basePath, ''), sidebarLinks);
 
+  const [open, setOpen] = useState(false);
+
+  const toggleOpen = () => setOpen(!open);
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:45]: setOpen(prevOpen => !prevOpen)
also add useCallback with [] deps array

**Review by polubis — CHANGES_REQUESTED** [review]

File: src/modules/main/Main.tsx (hunk 2 above)
Comment (ajronn) [reply src/modules/main/Main.tsx]: Sidebar is rendered by overlaying

File: src/modules/main/sidebar/Sidebar.tsx (hunk 3 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Logic has been moved inside the component

File: src/modules/main/sidebar/Sidebar.tsx (hunk 4 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Inline styles  have been replaced with styles in the .scss file

File: src/modules/main/sidebar/Sidebar.tsx (hunk 5 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Inline styles have been replaced with styles in the .scss file

File: src/modules/main/sidebar/Sidebar.tsx (hunk 6 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Inline styles have been replaced with styles in the .scss file

File: src/modules/main/sidebar/Sidebar.tsx (hunk 7 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Inline styles have been replaced with styles in the .scss file

File: src/modules/main/sidebar/Sidebar.tsx (hunk 8 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Inline styles have been replaced with styles in the .scss file

File: src/modules/main/sidebar/Sidebar.tsx (hunk 9 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Names have been changed

Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Callback removed

File: src/modules/main/sidebar/Sidebar.scss (hunk 10 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.scss:29]: Reverted changes

File: src/modules/main/sidebar/Sidebar.tsx (hunk 11 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx:14]: useState has been moved

File: src/modules/main/sidebar/Sidebar.tsx (hunk 12 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: Name has been changed

File: src/modules/main/sidebar/portal/Portal.scss (hunk 13 above)
Comment (ajronn) [reply src/modules/main/sidebar/portal/Portal.scss]: Name has been changed

File: src/modules/main/sidebar/portal/Portal.scss (hunk 14 above)
Comment (ajronn) [reply src/modules/main/sidebar/portal/Portal.scss]: Style has been changed

File: src/modules/main/sidebar/portal/Portal.tsx (hunk 15 above)
Comment (ajronn) [reply src/modules/main/sidebar/portal/Portal.tsx]: Has been fixed

File: src/modules/main/sidebar/portal/Portal.tsx (hunk 16 above)
Comment (ajronn) [reply src/modules/main/sidebar/portal/Portal.tsx]: Names have been changed

File: src/modules/main/sidebar/portal/Portal.tsx (hunk 17 above)
Comment (ajronn) [reply src/modules/main/sidebar/portal/Portal.tsx]: Component has beed created

File: src/modules/main/sidebar/portal/index.ts (hunk 18 above)
Comment (ajronn) [reply src/modules/main/sidebar/portal/index.ts]: Name has been changed

File: src/modules/main/sidebar/sidebar-panel/SidebarPanel.tsx [hunk 20]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+
+import { Logo } from 'ui';
+import Cancel from '@material-ui/icons/Cancel';
```
Comment (polubis) [inline src/modules/main/sidebar/sidebar-panel/SidebarPanel.tsx:4]: Material ui things should be moved between react import and Logo from UI - remember about enter between both

File: src/modules/main/sidebar/Sidebar.scss [hunk 21]
```diff
@@ -14,41 +14,8 @@
...
-
-      .marker {
-        position: absolute;
-        right: -2px;
-        width: 4px;
-        background: $secondary;
-        border-radius: 39px;
-        transition: transform 0.3s;
-      }
+    .tile{
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.scss:17]: This is not needed imo

File: src/modules/main/sidebar/Sidebar.tsx [hunk 22]
```diff
@@ -1,75 +1,41 @@
-import React from 'react';
-import { NavLink, useLocation } from 'react-router-dom';
+import React, {useState, useCallback} from 'react';
 
+import { Logo } from 'ui';
 import { Button } from '@material-ui/core';
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:4]: Enter - check readme file and section import order - will be much easier

File: src/modules/main/sidebar/Sidebar.tsx [hunk 23]
```diff
@@ -1,75 +1,41 @@
...
-    return pathname.includes(path);
-  });
-};
-
 export const Sidebar = ({ basePath }: SidebarProps) => {
-  const { pathname } = useLocation();
+  const [open, setOpen] = useState(false);
 
-  const activeLinkIdx = getActiveLinkIdx(pathname.replace(basePath, ''), sidebarLinks);
+  const toggleOpen = () => {
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:16]: wrap toggleOpen with useCallback statement with empty deps array

Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:16]: replace setOpen(!open) - with setOpen(prevOpen => !prevOpen) - setState fn is async

File: src/modules/main/sidebar/Sidebar.tsx [hunk 24]
```diff
@@ -1,75 +1,41 @@
...
-          <span
-            className={csx.marker}
-            style={{
-              height: `${MARKER_HEIGHT}px`,
-              top: `${(LINK_HEIGHT - MARKER_HEIGHT) / 2}px`,
-              transform: `translateY(${LINK_HEIGHT * activeLinkIdx}px)`
-            }}
-          />
-        </div>
+        {open && <SidebarPanel basePath={basePath} close={toggleOpen}/>}
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:33]: close - should be named onClose

File: src/modules/main/sidebar/Sidebar.tsx [hunk 25]
```diff
@@ -1,75 +1,41 @@
...
+        
+        <SidebarLinks basePath={basePath} component={(icon,label) => link(icon,label)}/>
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:35]: replace (icon, label) => (link(icon, label) - to renderLink - it will work anyawy

File: src/modules/main/sidebar/Sidebar.tsx [hunk 26]
```diff
@@ -1,75 +1,41 @@
...
-  const { pathname } = useLocation();
+  const [open, setOpen] = useState(false);
 
-  const activeLinkIdx = getActiveLinkIdx(pathname.replace(basePath, ''), sidebarLinks);
+  const toggleOpen = () => {
+      setOpen(!open);
+    }
+    
 
+  const link = (icon, label) => {
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:21]: rename to render link and add here useCallback with empty deps array

File: src/modules/main/sidebar/Sidebar.tsx [hunk 27]
```diff
@@ -1,75 +1,41 @@
...
+  return(<Button>{icon}<div className={csx.tile}>{label}</div></Button>)
```
Comment (polubis) [inline src/modules/main/sidebar/Sidebar.tsx:22]: <Button> should have class - csx.link - and tile class should be removed.

File: src/modules/main/sidebar/sidebar-links/SidebarLinks.scss [hunk 28]
```diff
@@ -0,0 +1,40 @@
...
+        padding: 0;
+        border-radius: 0;
+        color: $primary;
+
+        & > span {
+          @include col;
+
+          & > span {
+            margin-top: 6px;
+            @include linkSmall;
```
Comment (polubis) [inline src/modules/main/sidebar/sidebar-links/SidebarLinks.scss:20]: @include always should be first

File: src/modules/main/sidebar/sidebar-links/SidebarLinks.tsx [hunk 29]
```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { NavLink, useLocation } from 'react-router-dom';
+
+
```
Comment (polubis) [inline src/modules/main/sidebar/sidebar-links/SidebarLinks.tsx:4]: Remove enter

**Review by polubis — CHANGES_REQUESTED** [review]
Good job. Right now is much better. But still, there are some things to improve.
- spacing between links content and typography has been changed - for reference check this link,
- code formatting is bad. Every file you changed should be formatted with shortcut shift + alt + f in Vs code

File: src/modules/main/sidebar/Sidebar.tsx (hunk 19 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx]: The solution has been applied

File: src/modules/main/sidebar/sidebar-panel/SidebarPanel.tsx (hunk 20 above)
Comment (ajronn) [reply src/modules/main/sidebar/sidebar-panel/SidebarPanel.tsx:4]: Formatting has been improved

File: src/modules/main/sidebar/Sidebar.tsx (hunk 22 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx:4]: Formatting has been improved

File: src/modules/main/sidebar/Sidebar.tsx (hunk 23 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx:16]: The solution has been applied

Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx:16]: The solution has been applied

File: src/modules/main/sidebar/Sidebar.tsx (hunk 25 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx:35]: The solution has been applied

File: src/modules/main/sidebar/Sidebar.tsx (hunk 26 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx:21]: The solution has been applied

File: src/modules/main/sidebar/Sidebar.tsx (hunk 27 above)
Comment (ajronn) [reply src/modules/main/sidebar/Sidebar.tsx:22]: .link class is not in this scope. It was moved into SideBarLinks.scss file.

File: src/modules/main/sidebar/sidebar-links/SidebarLinks.scss (hunk 28 above)
Comment (ajronn) [reply src/modules/main/sidebar/sidebar-links/SidebarLinks.scss:20]: This code is not my, I transferred it from another file, but I fixed it

File: src/modules/main/sidebar/sidebar-links/SidebarLinks.tsx (hunk 29 above)
Comment (ajronn) [reply src/modules/main/sidebar/sidebar-links/SidebarLinks.tsx:4]: Formatting has been improved
